lib/daru/dataframe.rb

```diff
@@ -2006,6 +2005,18 @@ def split_by_category cat_name
 
     private
 
+    def headers
+      if index.is_a?(MultiIndex)
+        Daru::Index.new(index.name) | @vectors
```

Comment:
Wrong operator. It is "join of sets", so you'll have unexpected results if indexes and vectors have same names (possible for large stats datasets):
```ruby
v1 = Daru::Index.new(['month', 'year'])
v2 = Daru::Index.new(['salary', 'month', 'name'])
v1 | v2 # you'd like "month, year, salary, month, name" headers here, but...
# => #<Daru::Index(4): {month, year, salary, name}> 
```

I believe `Array(index.name) + @vectors.to_a` should be enough.